### PR TITLE
Feat/sentry plugin migration v1

### PR DESCRIPTION
**Pull request by scefali**
This PR moves over the test and source files from sentry_plugins into sentry. I have updated conftest.py file so plugins are installed before plugin tests (but won't install for non-plugin tests). Plugin tests work but won't run by default. Travis will run the tests as well.

Copied sentry-plugins from sha 5b3cd16629f0866d57bbf9d92ce3518bff8eb1dd

### CODE REVIEW

**Comment by joshuarli** [comment]
@scefali What sentry-plugins sha did you copy over?

**Comment by scefali** [comment]
> @scefali What sentry-plugins sha did you copy over?

@joshuarli  from 5b3cd16629f0866d57bbf9d92ce3518bff8eb1dd
Updated overview with this info as well

**Review by joshuarli — CHANGES_REQUESTED** [review]
@scefali You're going to have to refresh your dev env: `SENTRY_LIGHT_BUILD=1 pip install --no-use-pep517 -e '.[dev]'`

Then you'll start seeing things like `AssertionError: Missing plugin from entry_points: 'pushover'` and some dependency issues. You need to:

- copy over the setuptools entry points from sentry-plugins
- add sentry-plugins specific requirements to `requirements-base.txt`

Note that after you make these changes, you'll need to refresh your dev env again to install those entry points and the new dependencies.

**Comment by joshuarli** [comment]
@scefali I just pushed my local dev branch for reference:

- https://github.com/getsentry/sentry/commit/214ab97992cef9487b9057c67995730277d0e47e
- https://github.com/getsentry/sentry/commit/7f3cc06e5136b912f460e4cdcbd656d4f90c6cef

File: .travis.yml [hunk 1]
```diff
@@ -164,6 +164,11 @@ matrix:
       name: 'Django 1.9 Acceptance'
       env: DJANGO_VERSION=">=1.9,<1.10" TEST_SUITE=acceptance USE_SNUBA=1
 
+    # allowed to fail
```
Comment (joshuarli) [inline .travis.yml]: You're going to have to add in item with the exact env to `allow_failures` further below the travis config.

**Review by joshuarli — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (scefali) [reply .travis.yml]: @joshuarli Kewl, that's exactly what I was trying to figure out :)

File: requirements-base.txt [hunk 2]
```diff
@@ -1,6 +1,9 @@
 BeautifulSoup>=3.2.1
 boto3>=1.4.1,<1.4.6
 botocore<1.5.71
+
+# sentry-plugins specific dependencies
```
Comment (joshuarli) [inline requirements-base.txt]: oof, the fix imports pre-commit hook really likes to do this, I'd just `commit -n` this

**Comment by joshuarli** [comment]
Ah yeah, there's some acceptance tests inside of sentry-plugins, so you'll need `acceptance_default` for chromedriver. I feel like it's fine to just do that rather than run plugins acceptance separately in another test matrix.

File: src/sentry_plugins/github/testutils.py [hunk 3]
```diff
@@ -16,11 +16,11 @@
       "id": "133d60480286590a610a0eb7352ff6e02b9674c4",
       "tree_id": "f9d2a07e9488b91af2641b26b9407fe22a451433",
       "distinct": true,
-      "message": "Update README.md (àgain)",
+      "message": "Update README.md (again)",
```
Comment (joshuarli) [inline src/sentry_plugins/github/testutils.py]: `b"""` isn't a raw string, it's a bytestring. The comment above says to keep as a raw string... _shrugs_. Could you change that to `r"""` and undo the changes to `à`? I bet the linter won't complain.

**Review by joshuarli — CHANGES_REQUESTED** [review]

**Comment by wedamija** [comment]
You'll need to have ef0d63002ae730fb56bfbbc8a48d8a077d074244 from sentry-plugins included as well, I added that as part of Django migrations, and you'll end up with migration errors without the changes there. It basically just adds a Django migration for `jira_ac`.

**Comment by scefali** [comment]
@wedamija I think I have the changes from commit ef0d63002ae730fb56bfbbc8a48d8a077d074244. It looks to be the same commit as the one I listed: https://github.com/getsentry/sentry-plugins/commit/ef0d63002ae730fb56bfbbc8a48d8a077d074244

Maybe the different sha is from the squash and merge?

**Comment by scefali** [comment]
@wedamija Copied over the new migration

**Review by wedamija — APPROVED** [review]
lgtm, and staging is good too.

**Review by joshuarli — APPROVED** [review]
lgtm! Thanks @scefali.

File: setup.py [hunk 4]
```diff
@@ -138,7 +138,35 @@ def run(self):
...
     include_package_data=True,
-    entry_points={"console_scripts": ["sentry = sentry.runner:main"]},
+    entry_points={
+        "console_scripts": ["sentry = sentry.runner:main"],
+        "sentry.new_apps": [
+            "jira_ac = new_sentry_plugins.jira_ac",
+            "jira = new_sentry_plugins.jira",
+            "sessionstack = new_sentry_plugins.sessionstack",
+        ],
+        "sentry.new_plugins": [
```
[the remark this answers is not in the document]
Comment (BYK) [reply setup.py]: Isn't that the case already? https://github.com/getsentry/sentry/pull/15249

[the remark this answers is not in the document]
Comment (wedamija) [reply setup.py]: @BYK TIL. So it looks like when we finish sentry-plugins there will be some work on your end as well? Do you need us to coordinate with you?

[the remark this answers is not in the document]
Comment (BYK) [reply setup.py]: Yes please. Essentially when @scefali makes the final merge, we should revert/undo what we did in #15249.
